### PR TITLE
Add support for reloadDocument prop

### LINKED ISSUE #241: [Bug]: Not supporting <ValidatedForm reloadDocument>
### Which packages are impacted?

- [X] `remix-validated-form`
- [ ] `@remix-validated-form/with-zod`
- [ ] `@remix-validated-form/with-yup`
- [ ] `zod-form-data`

### What version of these packages are you using?

```
"@remix-validated-form/with-zod": "^2.0.5",
"remix-validated-form": "^4.6.7",
"zod-form-data": "^1.2.4"
```

### Please provide a link to a minimal reproduction of the issue.

https://github.com/Mange/remix-redirect-repro

### Steps to Reproduce the Bug or Issue

1. `npm install && npm run dev`
2. Go to http://localhost:3000
3. Test the different ways to submit.

### Expected behavior

Submitting a `<ValidatedForm reloadDocument>` should work the same was as a `<Form reloadDocument>`.

### Screenshots or Videos

image

## All but the Form with reloadDocument

image

## Form with reloadDocument

image

### Platform

- OS: Linux (Not relevant)
- Browser: Brave (Not relevant)
- Version: Not relevant

### Additional context

This breaks in `ValidatedForm` because it uses `useSubmit` under the hood, which does not respect `reloadDocument` or (AFACT) offers similar functionality.

I might not be diagnosing this issue correctly. It's all a guess from me.

I think a `reloadDocument` option should make it submit for a full page submit, like `noValidate` does, when JS is not available, but still submit the `<form>` for real without `useSubmit` if validations pass when JS is available.

Overall, I only want this behavior due to the lack of "hard redirects" in Remix, where I have some non-remix routes that I want to redirect to. If this is ever fixed upstream (remix-run/react-router#9859), then I won't care about this anymore.

### PR DESCRIPTION
**Pull request by JosiahSayers**
Fixes #241

### CODE REVIEW

File: packages/remix-validated-form/src/ValidatedForm.tsx [hunk 1]
```diff
@@ -364,18 +365,23 @@ export function ValidatedForm<DataType>({
...
-      onSubmit={(e) => {
-        e.preventDefault();
-        handleSubmit(
-          e,
-          e.currentTarget,
-          (e as unknown as HTMLSubmitEvent).nativeEvent
-        );
-      }}
+      onSubmit={
+        reloadDocument
```
Comment (airjp73) [inline packages/remix-validated-form/src/ValidatedForm.tsx:374]: As-is, this is going to skip all validations, which I'm not sure is desirable. Shouldn't we validate, then submit the form via a full page reload? Not sure off the top of my head how to handle that situation when there's a `submitter` other than the form though.

Comment (JosiahSayers) [reply packages/remix-validated-form/src/ValidatedForm.tsx:374]: That’s a good point. I was going after this with the thought process that if you wanted a full page reload you would be okay with the server handling validation, like when JS isn’t running clientside. I can see how that’s not a super desirable solution though.

Comment (machour) [reply packages/remix-validated-form/src/ValidatedForm.tsx:374]: Just hit this "bug" while upgrading from Remix 1.8, to 1.19.0.
Seems like something changed in the framework (probably when it was rewritten on top of react-router), that my use case fail:

`<ValidatedForm reloadDocument>` on a regular route, posting to a resource route that generates and download a PDF.

Used `patch-package` to implement this PR approach, but would definitely love something that keeps client-side validation.

Anything I can do to help this PR move forward?

**Comment by Mange** [comment]
Is there anything I can help you with to get this out faster?

**Comment by JosiahSayers** [comment]
> Is there anything I can help you with to get this out faster?

This fell off my radar as things have gotten busy for me. I don't really have the time to work on a fix at the moment, anyone who does can feel free to pick it up.

**Comment by airjp73** [comment]
Hi! This feature is implemented in the upcoming RVF v6, so I'm going to close this PR now. Thank you again for the time you put into it!

**Pull request closed without merging**